### DISCUSSION TITLE
Add ability to dry-run the deploy action

### DISCUSSION THREAD

**Discussion by Pike** [JamesIves/github-pages-deploy-action#499] [Ideas]
I'm currently working on a PR to add this action to deploy to gh-pages.

I noticed that there's no way in the PR to see how this step would perform, if it was run with sufficient perms and on master.

It'd be nice to have a dry-run mode, which would do anything but `git push`, and do a `git show HEAD` instead, maybe?

**Comment by JamesIves** [comment]
I like the idea of this. Technically there's a `--dry-run` flag for `git push` so that flag can optionally be included there too, allowing the entire workflow to finish. 

This is a good candidate for a version 4 feature release.

**Reply by Pike** [reply]
Thanks. I'm tempted to start contributing, but I haven't hacked on actions itself yet.

`CONTRIBUTING.md` mentions the master branch? It seems you stopped using master?

I'm also trying to wrap my head around the testing story. One thing that surprised me was the `v3-test` tests. And on the integration tests, I wonder if it could/should `uses: ./` instead of the published action?

**Reply by JamesIves** [reply]
You need to point it to your forked default branch (Which I think defaults to master?). For testing there's unit tests setup with Jest, and the integration tests are pointed to the release branch because you're required to commit dependencies. If there was a way to point it directly without committing node_modules to the dev branch I'd be all for it. 

I'd be happy to help guide you through any questions you have if you'd like to make contributions!

**Reply by Pike** [reply]
Just created a clone, it creates a default dev branch, https://github.com/Pike/github-pages-deploy-action.

I'll play with the integration tests as a first thing, I want to see if it works with having a lib and node_modules dir created at build time.

**Reply by JamesIves** [reply]
Love that, thank you! I actually found this today, it may be helpful: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#example-using-action-in-the-same-repository-as-the-workflow

I've also updated the contribution guide to include the reference to `@dev` instead of `@master`. With the release of version 4 the master branch will be entirely removed as it's running version 1.
